pkg/services/cosmosdb/common-provision.go

```diff
@@ -57,11 +94,58 @@ func (c *cosmosAccountManager) preProvision(
 }
 
 func (c *cosmosAccountManager) buildGoTemplateParams(
+	pp *provisioningParameters,
 	dt *cosmosdbInstanceDetails,
+	kind string,
 ) map[string]interface{} {
 	p := map[string]interface{}{}
 	p["name"] = dt.DatabaseAccountName
-	p["kind"] = "GlobalDocumentDB"
+	p["kind"] = kind
+
+	filters := []string{}
+	if pp.IPFilterRules != nil {
+		if pp.IPFilterRules.AllowAzure != "disable" {
```

Comment:
When you validated this, that check was case insensitive because you downcased input before comparing, but here you haven't done that. This could lead to some surprising behavior.

Reply:
Oooooh! Will fix!